sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/caches/AsyncCacheNonBlocking.java

```diff
@@ -73,12 +75,21 @@ public Mono<TValue> getAsync(
         Function<TValue, Mono<TValue>> singleValueInitFunc,
         Function<TValue, Boolean> forceRefresh) {
 
+        return Mono.create(sink ->
+            Mono.fromFuture(CompletableFuture.runAsync(() -> getAsyncInternal(key, singleValueInitFunc, forceRefresh)
+                .subscribeOn(CosmosSchedulers.ASYNC_CACHE_BACKGROUND_REFRESH_BOUNDED_ELASTIC)
```

Comment:
worth doing a perf testing

Reply:
@mbhaskar  / @jeet1995  - this looks like the same change that we have done in ProactiveConnectionManagement PR as well, can we double check the code here with that PR, also may be worth including in a helper method, if we can.